lib/assets/javascripts/cartodb/table/menu_modules/filters/selection.js

```diff
@@ -316,10 +318,26 @@
           }.bind(this)
         })
         .data("ui-autocomplete")._renderItem = function( ul, item ) {
-          var a = null;
+          var value = "<a>" + item.label + " <span style=\"float:right\">" + item.value  + "</span></a>";
+          var idx = -1;
+          if (typeof item.label === 'string' && typeof search_term === 'string') {
+            idx = item.label.toLowerCase().indexOf(search_term.toLowerCase());
+          }
+          if (idx !== -1) {
+            var tmpLabel = item.label;
+            value = [
+              "<a>", 
+              tmpLabel.slice(0, idx), 
+              '<span class="highlight" style="background-color: #ff9e24;">', 
+              tmpLabel.slice(idx, idx + search_term.length), 
+              "</span>", 
+              tmpLabel.slice(idx + search_term.length, tmpLabel.length), 
+              " <span style=\"float:right\">" + item.value  + "</span></a>"
+            ].join('');
+          }
```

Comment:
Couldn't this algorithm be rewritten as:

```js
var index = -1;
if (typeof item.label === 'string' && typeof search_term === 'string') {
    index = item.label.toLowerCase().indexOf(search_term.toLowerCase());
}

var value;
if (index === -1) {
    value = "<a>" + item.label + " <span style=\"float:right\">" + item.value  + "</span></a>";
} else {
    var tmpLabel = item.label;
    value = [
        "<a>", 
        tmpLabel.slice(0, index), 
        '<span class="highlight" style="background-color: #ff9e24;">', 
        tmpLabel.slice(index, index + search_term.length), 
        "</span>", 
        tmpLabel.slice(index + search_term.length, tmpLabel.length), 
        " <span style=\"float:right\">" + item.value  + "</span></a>"
    ].join('');
}
```
 It's just reorganising the code a bit, but IMHO this way it has some advantages:
1. The reader processes the calculation of the index first (with all the typeof complexity) and then is left to tackle the complexity of the value part. The way you presented it originally, it's not clear that both are independent, and the reader tries to draw straws from one to another. Their independent (as far as I can tell) so it's best to free the reader from the doubt when first approaching the code.
2. The if statement can now start with an affirmation (a === b) instead of a negation (a !== b), which is always harder to read.

Give some thought to it and change it if you agree!

Reply:
I agree that is easier to read and I have made the adjustments.